pyrsia_node/src/metadata_manager/metadata.rs

```diff
@@ -172,3 +186,462 @@ impl Iterator for PackageIterator {
+            warn!(
+                "Failed to open document store {}; error was {}",
+                ds_name, error
+            );
+            info!("Attempting to create document store {}", ds_name);
+            match DocumentStore::create(ds_name, index_specs()) {
+                Ok(ds) => {
+                    info!("Created document store {}", ds_name);
+                    populate_with_initial_records(&ds, initial_records)?;
+                    Ok(ds)
+                }
+                Err(error) => failed_to_create_document_store(ds_name, error),
+            }
+        }
+    }
+}
+
+fn populate_with_initial_records(
+    ds: &DocumentStore,
+    initial_records: fn() -> Vec<String>,
+) -> Result<()> {
+    for record in initial_records() {
+        if let Err(error) = ds.insert(&record) {
+            error!(
+                "Failed to insert initial record into document store {}: {}\nError was {}",
+                ds.name, record, error
+            );
+            todo!("If an attempt to insert an initial record into document store fails, then we need to do something so that we will know that the document store is missing necessary information")
+        }
+    }
+    Ok(())
+}
+
+fn failed_to_create_document_store(
+    ds_name: &str,
+    error: Box<dyn Error>,
+) -> Result<DocumentStore, anyhow::Error> {
+    let msg = format!(
+        "Failed to create document store {} due to error {}",
+        ds_name, error
+    );
+    error!("{}", msg);
+    return Err(anyhow!(msg));
+}
+
+impl MetadataApi for Metadata<'_> {
+    fn create_package_type(&self, pkg_type: &PackageType) -> anyhow::Result<(), anyhow::Error> {
+        match self.trust_manager.trust_package_type(pkg_type) {
+            Ok(_) => insert_metadata(&self.package_type_docs, pkg_type),
+            Err(error) => untrusted_metadata_error(pkg_type, &error.to_string()),
+        }
+    }
+
+    fn get_package_type(
+        &self,
+        name: PackageTypeName,
+    ) -> anyhow::Result<Option<PackageType>, anyhow::Error> {
+        let name_as_string = name.to_string();
+        let filter = hashmap! {
+            FLD_PACKAGE_TYPES_NAME => name_as_string.as_str()
+        };
+        match self.package_type_docs.fetch(IX_PACKAGE_TYPES_NAMES, filter) {
+            Err(error) => Err(anyhow!("Error fetching package type: {}", error)),
+            Ok(Some(json)) => Ok(Some(PackageType::from_json_string(&json)?)),
+            Ok(None) => Ok(None),
+        }
+    }
+
+    fn create_namespace(&self, namespace: &Namespace) -> anyhow::Result<(), anyhow::Error> {
+        match self.trust_manager.trust_namespace(namespace) {
+            Ok(_) => insert_metadata(&self.namespace_docs, namespace),
+            Err(error) => untrusted_metadata_error(namespace, &error.to_string()),
+        }
+    }
+
+    fn get_namespace(
+        &self,
+        package_type: PackageTypeName,
+        namespace_path: &str,
+    ) -> anyhow::Result<Option<Namespace>, anyhow::Error> {
```

Comment:
According to the `anyhow` docs this is the same thing.

```suggestion
    ) -> anyhow::Result<Option<Namespace>> {
```

Is there a reason you choose to add the default error type in?

Reply:
No. I should simplify the declarations as there is no advantage to the more complex form for the default case. Thank you for pointing this out.